rustls/src/server/server_conn.rs

```diff
@@ -136,6 +144,8 @@ impl<'a> ClientHello<'a> {
         server_name: &'a Option<DnsName<'_>>,
         signature_schemes: &'a [SignatureScheme],
         alpn: Option<&'a Vec<ProtocolName>>,
+        server_cert_type: Option<&'a [CertificateType]>,
+        client_cert_type: Option<&'a [CertificateType]>,
```

Comment:
Is there perhaps a better way to represent this so that the server/client types can't be swapped accidentally?

Reply:
I've added a `Server/ClientCertType` tuple struct to `server_conn.rs`, I think it's more robust like this. Or do you have a different method in mind? 
Also, what would be a good location to add these tuple structs?

Reply:
Your approach looks good to me and I think where you defined the struct makes sense :+1:  Thanks!